user_tools/src/spark_rapids_pytools/rapids/rapids_job.py

```diff
@@ -218,3 +224,32 @@ def _submit_job(self, cmd_args: list) -> str:
         out_std = self.exec_ctxt.platform.cli.run_sys_cmd(cmd=cmd_args,
                                                           env_vars=env_args)
         return out_std
+
+
+@dataclass
+class RapidsDistributedJob(RapidsJob):
+    """
+    Implementation of a RAPIDS job that runs distributed on a cluster.
+    """
+
+    def _build_submission_cmd(self) -> JarCmdArgs:
+        classpath_arr = self._build_classpath()
+        hadoop_cp = self._get_hadoop_classpath()
+        jvm_args_arr = self._build_jvm_args()
+        jar_main_class = self.prop_container.get_jar_main_class()
+        jar_output_dir_args = self._get_persistent_rapids_args()
+        extra_rapids_args = self.prop_container.get_rapids_args()
+        return JarCmdArgs(jvm_args_arr, classpath_arr, hadoop_cp, jar_main_class,
```

Comment:
QQ: what is the reason behind adding this `JarCmdArgs` class? Why not keeping it the same as for local job and just using `list`?

Reply:
For distributed job, we need this wrapper case class `JarCmdArgs` as each map task is going to recreate the java command with new arguments. It cannot use the list because there are some arguments that need to assigned at runtime (e.g. location of Tools JAR).

Reply:
thanks @parthosa!